src/machine/machinectl.c

```diff
@@ -1236,6 +1236,40 @@ static int process_forward(sd_event *event, PTYForward **forward, int master, PT
         return ret;
 }
 
+static int parse_machine_uid(const char *spec, const char **machine, char **uid) {
+        /*
+         * Whatever is specified in the spec takes priority over global arguments.
+         */
+        char *_uid;
+        const char *_machine;
+
+        if (spec) {
+                const char *at;
+
+                at = strchr(spec, '@');
+                if (at) {
+                        _uid = strndup(spec, at - spec);
+                        _machine = at + 1;
+                } else {
+                        _uid = strdup(arg_uid);
+                        _machine = spec;
+                }
+
+                if (isempty(_machine))
+                        _machine = ".host";
+        } else {
+                _uid = strdup(arg_uid);
+                _machine = ".host";
+        }
```

Comment:
btw, instead of splitting this manually, maybe we should actually use extract_first_word() to split the left hend out, and then the remainder is the machine name...